js/viewer/annotation/base.js

```diff
@@ -81,15 +58,27 @@ function Base(attrs){
                 case "graph-id":
```

Comment:
You should figure out why there's `tag` and `graph-id` in attributes. If these are values that you need in the object, they shouldn't be part of `_attrs` and instead should be separate properties of your class.

Reply:
If you plan on leaving the `graph-id`, you should at least get rid of this useless switch case. You can just add `attr === "tag" || attr === "graph-id"` to the if statement above.

Reply:
Done